unittests/osr_test.py

```diff
@@ -0,0 +1,153 @@
+              print('mousemove:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("dragenter", function( e ) {
+              print('dragenter:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("dragover", function( e ) {
+              print('dragover:'+e.pageX+","+e.pageY)
+            }, false);
+            document.addEventListener("drop", function( e ) {
+              print('drop:'+e.pageX+","+e.pageY)
+            }, false);
+
+
+        }
+        print('window.onload init')
+    </script>
+    <style>
+        .box{
+            background-color: #F00;
+            height: 200px;
+        }
+    </style>
+</head>
+<body>
+
+<input id="selectFile" type="file">
+<div class="box"></div>
+</body>
+</html>
+
+"""
+
+g_datauri = "data:text/html;base64," + base64.b64encode(g_datauri_data.encode(
+    "utf-8", "replace")).decode("utf-8", "replace")
+VIEWPORT_SIZE = (1024, 768)
+
+g_subtests_ran = 0
+
+def subtest_message(message):
+    global g_subtests_ran
+    g_subtests_ran += 1
+    print(str(g_subtests_ran) + ". " + message)
+    sys.stdout.flush()
+
+class Handler(object):
+    def __init__(self):
+        self.OnPaint_called = False
+
+    def OnConsoleMessage(self, message, **_):
+        if "error" in message.lower() or "uncaught" in message.lower():
+            self.javascript_errors_False = True
+            raise Exception(message)
+        else:
+            # Check whether messages from javascript are coming
+            self.OnConsoleMessage_True = True
+            subtest_message(message)
+
+    def OnDragEnter(self, browser, dragData, mask):
+        subtest_message('cef.OnDragEnter() OK')
+        return False
+
+    def StartDragging(self,browser, drag_data, allowed_ops, x, y):
+        subtest_message('cef.StartDragging() OK')
+        return False
+
+    def OnLoadEnd(self,browser,frame,http_code):
+        subtest_message('cef.OnLoadEnd() OK')
+        #testing mouse event
+        browser.SendMouseClickEvent(300, 20, cef.MOUSEBUTTON_LEFT, False, 1, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
+        browser.SendMouseMoveEvent(305, 25, False, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
+        browser.SendMouseClickEvent(305, 25, cef.MOUSEBUTTON_LEFT, True, 1, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
+
+        # testing drag event
+        # It can trigging cef.OnDragEnter, but still not trigger dragenter event in JS.
+        dragData = cef.DragData()
+        dragData.SetFragmentText('Test')
+        dragData.ResetFileContents()
+        browser.DragTargetDragEnter(dragData, 301, 21, cef.DRAG_OPERATION_COPY )
+        browser.DragTargetDragOver(302, 22, cef.DRAG_OPERATION_COPY )
+        browser.DragTargetDrop(303, 23)
+        browser.DragSourceEndedAt(303, 23, cef.DRAG_OPERATION_COPY )
+        browser.DragSourceSystemDragEnded()
+
+
+
+
+class OSRTest_IsolatedTest(unittest.TestCase):
+    def test_dragdrop(self):
```

Comment:
Just info for the future - with IsolatedTest you can have multiple separate tests in one test file and each of these isolated tests can initalize CEF and shutdown CEF, because they are run using separate processes. But I don't think we need to put these two tests (dragdata and dragdrop) as isolated tests, it would just increase the time to run both tests (CEF initalization, shutdown, message loop - it all takes time). If there is a case for it, when two tests need to be run in separate processes as they might affect each other, then sure go for it and create two isolated tests or more in one test file.